auth/server/src/main/java/com/seagates3/controller/AccountController.java

```diff
@@ -176,7 +179,23 @@ public ServerResponse create() {
 
         AccessKey rootAccessKey;
         try {
-            rootAccessKey = createRootAccessKey(root);
+            rootAccessKey = accessKeyDAO.find(accessKey);
+        } catch (DataAccessException ex) {
+          LOGGER.error("Failed to find access key in ldap -" + ex);
+            return accountResponseGenerator.internalServerError();
+        }
+
+        if (rootAccessKey.exists()) {
+            return accountResponseGenerator.accessKeyAlreadyExists();
+        }
+
+        try {
+            if (accessKey == null) {
+                rootAccessKey = AccessKeyService.createAccessKey(root);
```

Comment:
Looks like you have copied the accekey generation  function to AccessKeyService class
Kindly remove the method from this class and update its references everywhere.
This way we will avoid code duplication.

Reply:
I added this method in the service class as this logic is being used at multiple places. Also we planned to update saving secret key logic (encrypt/decrypt). Hence I had started the refactoring. Kindly let me know if you still think that I should remove it from AccessService class.

Reply:
I think you misunderstood.
I am telling to remove from AccountController. 
Keep everything under AccessService and update references everywhere.

Reply:
Will fix this.

Reply:
Done